docs/contrib-guide/profiling.md

```diff
@@ -1,5 +1,31 @@
 # Profiling
 
+## CPU profiling
+
+### Setup
+
+First you need to install `samply`. You can install it with:
+
+```bash
+cargo binstall samply
+```
+
+### Build
+
+To build Rolldown with the information required by `samply`, you need to build it with:
+
+```shell
+just build-profile
+```
+
+### Profiling
+
+After building, you can run Rolldown with the following command to profile CPU usage:
```

Comment:
Do we need to split this into two sections?
If a user wants to profile Rust code only, they could just do:
```bash
samply record node ./path/to/script-rolldown-is-used.js
```

If the user wants to profile both Rust and JavaScript parts (or JavaScript only), they could use:
```bash
samply record node --perf-prof --perf-basic-prof --perf-prof-unwinding-info --interpreted-frames-native-stack ./path/to/script-rolldown-is-used.js
```

The purpose of this is to avoid misleading users that profiling rolldown requires very long node cli flags.
Additionally, we could paste the Twitter URL so that they could know the reason for using long CLI flags.

Reply:
https://x.com/RaisinTen/status/1918297697532231954